src/diffusers/models/embeddings.py

```diff
@@ -790,3 +791,96 @@ def forward(self, caption, force_drop_ids=None):
         hidden_states = self.act_1(hidden_states)
         hidden_states = self.linear_2(hidden_states)
         return hidden_states
+
+
+class Resampler(nn.Module):
+    """Resampler of IP-Adapter Plus.
+
+    Args:
+    ----
+        embed_dims (int): The feature dimension. Defaults to 768.
+        output_dims (int): The number of output channels, that is the same
+            number of the channels in the
+            `unet.config.cross_attention_dim`. Defaults to 1024.
+        hidden_dims (int): The number of hidden channels. Defaults to 1280.
+        depth (int): The number of blocks. Defaults to 8.
+        dim_head (int): The number of head channels. Defaults to 64.
+        heads (int): Parallel attention heads. Defaults to 16.
+        num_queries (int): The number of queries. Defaults to 8.
+        ffn_ratio (float): The expansion ratio of feedforward network hidden
+            layer channels. Defaults to 4.
+    """
+
+    def __init__(
+        self,
+        embed_dims: int = 768,
+        output_dims: int = 1024,
+        hidden_dims: int = 1280,
+        depth: int = 4,
+        dim_head: int = 64,
+        heads: int = 16,
+        num_queries: int = 8,
+        ffn_ratio: float = 4,
+    ) -> None:
+        super().__init__()
+        self.latents = nn.Parameter(torch.randn(1, num_queries, hidden_dims) / hidden_dims**0.5)
+
+        self.proj_in = nn.Linear(embed_dims, hidden_dims)
+
+        self.proj_out = nn.Linear(hidden_dims, output_dims)
+        self.norm_out = nn.LayerNorm(output_dims)
+
+        self.layers = nn.ModuleList([])
+        for _ in range(depth):
+            self.layers.append(
+                nn.ModuleList(
+                    [
+                        nn.LayerNorm(hidden_dims),
+                        nn.LayerNorm(hidden_dims),
+                        Attention(
+                            query_dim=hidden_dims,
+                            dim_head=dim_head,
+                            heads=heads,
+                            out_bias=False,
+                        ),
+                        self._get_ffn(embed_dims=hidden_dims, ffn_ratio=ffn_ratio),
+                    ]
+                )
+            )
+
+    def _get_ffn(self, embed_dims, ffn_ratio=4) -> nn.Sequential:
+        """Get feedforward network."""
+        inner_dim = int(embed_dims * ffn_ratio)
+        return nn.Sequential(
+            nn.LayerNorm(embed_dims),
+            nn.Linear(embed_dims, inner_dim, bias=False),
+            nn.GELU(),
+            nn.Linear(inner_dim, embed_dims, bias=False),
+        )
```

Comment:
```suggestion
    def _get_ffn(self, embed_dims, ffn_ratio=4) -> nn.Sequential:
        """Get feedforward network."""
        inner_dim = int(embed_dims * ffn_ratio)
        return nn.Sequential(
            nn.LayerNorm(embed_dims),
            nn.Linear(embed_dims, inner_dim, bias=False),
            nn.GELU(),
            nn.Linear(inner_dim, embed_dims, bias=False),
        )
```

Can we replace this with a layer norm layer 
 first and then https://github.com/huggingface/diffusers/blob/7d4a257c7f2a8fa85fdc1532d19f7dce92e5a3d9/src/diffusers/models/attention.py#L493?

Reply:
`FeedForward` is `act` -> `linear`

If I replace it, it becomes `LayerNorm` -> `Linear` -> `FeedForward`. Is this necessary?